invisible_cities/detector_simulation/sensor_utils_test.py

```diff
@@ -0,0 +1,94 @@
+from pytest import    mark
+
+from .. core import system_of_units as units
+
+from . sensor_utils import   first_and_last_times
+from . sensor_utils import          get_n_sensors
+from . sensor_utils import          order_sensors
+from . sensor_utils import pmt_and_sipm_bin_width
+from . sensor_utils import          trigger_times
+
+
+def test_trigger_times():
+    event_time = 22 * units.ns
+    bin_width  = 0.1
+    time_bins  = np.arange(0, 10, bin_width)
+    triggers   = [15, 75]
+
+    trgr_times = trigger_times(triggers, event_time, time_bins)
+    assert len(trgr_times) == len(triggers)
+    assert trgr_times[0]   == event_time + triggers[0] * bin_width
+    assert trgr_times[1]   == event_time + triggers[1] * bin_width
+
+
+def test_first_and_last_times():
+    bin_min    = 22
+    bin_max    = 40
+    test_bins1 = np.arange(bin_min, bin_max)
+    test_bins2 = np.arange(bin_min, bin_max - 1)
+
+    bmin, bmax = first_and_last_times(pd.Series({'time': test_bins1}),
+                                      pd.Series({'time': test_bins2}),
+                                      1, 1)
+
+    assert np.round(bmin) == bin_min
+    assert np.round(bmax) == bin_max
+
+@fixture(scope = 'function')
+def id_dict():
+    id_dict = {'pmt_ids' :    (2, 5, 7), 'pmt_ord' :  (2, 5, 7),
+               'sipm_ids': (1010, 5023), 'sipm_ord':  (10, 279)}
+    return id_dict
+def test_order_sensors(id_dict, pmt_ids, sipm_ids):
+    detector_db = 'new'
+    run_number  = 6400
+    n_pmt       = len( pmt_ids)
+    n_sipm      = len(sipm_ids)
+    nsamp_pmt   = 5000
+    nsamp_sipm  =    5
+
+    sensor_order = order_sensors(detector_db, run_number,
+                                 n_pmt      ,  nsamp_pmt,
+                                 n_sipm     , nsamp_sipm)
+
+    pmt_resp   = pd.Series([[1]*nsamp_pmt ]*len(id_dict[ 'pmt_ids']),
+                          index = id_dict[ 'pmt_ids'])
+    sipm_resp  = pd.Series([[1]*nsamp_sipm]*len(id_dict['sipm_ids']),
+                          index = id_dict['sipm_ids'])
+    pmt_q      = np.array( pmt_resp.tolist())
+    sipm_q     = np.array(sipm_resp.tolist())
+    padded_evt = sensor_order(pmt_resp, sipm_resp, [(pmt_q, sipm_q)])
+
+    pmt_out  = padded_evt[0][0]
+    sipm_out = padded_evt[0][1]
+    assert pmt_out.shape == (n_pmt, nsamp_pmt)
+    pmt_nonzero = np.argwhere(pmt_out.sum(axis=1) != 0)
+    assert np.all(pmt_nonzero.flatten() == id_dict['pmt_ord'])
+    assert sipm_out.shape == (n_sipm, nsamp_sipm)
+    sipm_nonzero = np.argwhere(sipm_out.sum(axis=1) != 0)
+    assert np.all(sipm_nonzero.flatten() == id_dict['sipm_ord'])
+
+
+def test_get_n_sensors(pmt_ids, sipm_ids):
+    npmt, nsipm = get_n_sensors('new', 6400)
+
+    assert npmt  == len( pmt_ids)
+    assert nsipm == len(sipm_ids)
+
+
+def test_pmt_and_sipm_bin_width(ICDATADIR):
+    file_in = os.path.join(ICDATADIR                                         ,
+                           'Kr83_full_nexus_v5_03_01_ACTIVE_7bar_1evt.sim.h5')
```

Comment:
This filename is defined in the fixture `mc_waveforms`. It should probably be defined in an independent fixture to be reused here. Actually I've just seen that the same file is also used in other modules, so it should probably go in the global conftest.py file and be fixed later in the other modules.